### PR TITLE
[docker] container & volume metrics (wraps #2740)

### PR DESCRIPTION
**Pull request by hkaj**
Based on the great work of @parkr in https://github.com/DataDog/dd-agent/pull/2740

and 

- fix tests & conflicts
- change how we name and tag the metrics a little
- kill some dead code

### CODE REVIEW

**Comment by hkaj** [comment]
test failure unrelated

**Comment by hkaj** [comment]
@parkr yes i tested it locally. Testing it on travis would require sudo and docker, but to speed up tests we're using their container infrastructure which doesn't support sudo and DinD (wise decision, but painful here).

**Comment by parkr** [comment]
@hkaj Hey! Anything I can do to help get this past the finish line?

**Comment by parkr** [comment]
Tried this on a host with 5.10.1 installed:

```text
2017-01-09 10:50:12 PST | ERROR | dd.collector | checks.docker_daemon(__init__.py:784) | Check 'docker_daemon' instance #0 failed
Traceback (most recent call last):
  File "/opt/datadog-agent/agent/checks/__init__.py", line 767, in run
    self.check(copy.deepcopy(instance))
  File "/etc/dd-agent/checks.d/docker_daemon.py", line 255, in check
    self._report_performance_metrics(containers_by_id)
  File "/etc/dd-agent/checks.d/docker_daemon.py", line 580, in _report_performance_metrics
    self._report_cgroup_metrics(container, tags)
  File "/etc/dd-agent/checks.d/docker_daemon.py", line 598, in _report_cgroup_metrics
    stat_file = self._get_cgroup_from_proc(cgroup["cgroup"], container['_pid'], cgroup['file'])
  File "/etc/dd-agent/checks.d/docker_daemon.py", line 861, in _get_cgroup_from_proc
    return DockerUtil.find_cgroup_from_proc(self._mountpoints, pid, cgroup, self.docker_util._docker_root) % (params)
  File "/opt/datadog-agent/agent/utils/dockerutil.py", line 311, in find_cgroup_from_proc
    with open(proc_path, 'r') as fp:
IOError: [Errno 2] No such file or directory: '/proc/24207/cgroup'
```

**Comment by therc** [comment]
@parkr: did you set `docker_root: /host` in the init_config section of your docker_daemon.yaml? I just wasted a whole hour on this because I had started from a more recent template and erroneously assumed that it was the default setting.

**Comment by parkr** [comment]
> @parkr: did you set docker_root: /host in the init_config section of your docker_daemon.yaml? I just wasted a whole hour on this because I had started from a more recent template and erroneously assumed that it was the default setting.

No, I never set `docker_root`. In general, `/proc/$PID/cgroup` is the correct place to look, it's just that it doesn't always contain this file (depending on the process). Sometimes the proc quits before the cgroup info can be read. I think it's just a race condition.

**Comment by hkaj** [comment]
@hush-hush i rebased it, can you have a look please?

File: checks.d/docker_daemon.py [hunk 1]
```diff
@@ -540,14 +529,43 @@ def _submit_healthcheck_sc(self, container):
...
-                self.gauge('docker.image.size', image['Size'], tags=tags)
-
+            try:
+                tags = self._get_tags(image, IMAGE)
+                if 'VirtualSize' in image:
+                    self.gauge('docker.image.virtual_size', image['VirtualSize'], tags=tags)
+                if 'Size' in image:
+                    self.gauge('docker.image.size', image['Size'], tags=tags)
+            except:
+                import ipdb;ipdb.set_trace()
```
Comment (hush-hush) [inline checks.d/docker_daemon.py]: This should be on 2 line (that why the CI flake8 tests are falling). End I'm not sure we want to keep it, looks a lot like some debug lines.

File: tests/checks/integration/test_docker_daemon.py [hunk 2]
```diff
@@ -73,9 +73,9 @@ def mock_get_info_no_data(self):
     def mock_get_info_invalid_values(self):
         return {
             'DriverStatus': [
-                ['Metadata Space Available', '9 MB'],
-                ['Metadata Space Total', '10 MB'],
-                ['Metadata Space Used', '11 MB'],
+                ['Metadata Space Available', '0 MB'],
```
Comment (hush-hush) [inline tests/checks/integration/test_docker_daemon.py]: why do we set values to 0 ? Doesn't that broke `test_devicemapper_invalid_values` test ?

**Comment by parkr** [comment]
🎉  Thank you, all!

**Comment by hkaj** [comment]
Thank **you** @parkr  🍰

**Comment by parkr** [comment]
@hkaj Thank _you_ for following up and writing the tests. And for ensuring it got reviewed and merged eventually. 😄  Will this make it into a release in the coming week(s)?

**Comment by hkaj** [comment]
5.11 is dedicated to releasing the new APM agent so we won't release any change from dd-agent then, but 5.12 will follow soon after.

**Pull request merged**